code-challenges.test.js

```diff
@@ -17,45 +20,129 @@
+            expect(secretCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+            expect(secretCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+            expect(secretCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+        })
+    })
+
+// b) Create the function that makes the test pass.
+
+// Create a function named secretCode
+// That takes a string as an arguement for the input
+const secretCode = (string) => {
+// Use the replace method
+// Use regular expressions which allows you to use global replacement to replace multiple characters in a string
+// Use the "i" flag to make the replacement case insensitive
+// Chain the methods together
+// Return
+    return string.replace(/a/gi, 4).replace(/e/gi, 3).replace(/i/gi, 1).replace(/o/gi, 0)
+}
+
 var secretCodeWord1 = "Lackadaisical"
+console.log(secretCode(secretCodeWord1))
 // Expected output: "L4ck4d41s1c4l"
 var secretCodeWord2 = "Gobbledygook"
+console.log(secretCode(secretCodeWord2))
 // Expected output: "G0bbl3dyg00k"
 var secretCodeWord3 = "Eccentric"
+console.log(secretCode(secretCodeWord3))
 // Expected output: "3cc3ntr1c"
 
+// --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
 
+// a) Create a test with an expect statement using the variable provided.
 
-// b) Create the function that makes the test pass.
-
-
+// Describe the name of the function - aLetterFilter
+describe("aLetterFilter", () => {
+    // Test what the function does - returns an array of all the words of a given array that contain the letter a
+        it("returns an array of all the words of a given array that contain the letter a", () => {
+    // Expect the output of the function to be equal to the correct output - output of aLetterFilter to equal an array of all the words of a given array that contain the letter a
+            expect(aLetterFilter(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"])
+            expect(aLetterFilter(arrayOfWords2)).toEqual(["Mango", "Apricot", "Peach"])
+        })
+    })
 
-// --------------------2) Create a function that takes in an array and returns all the words that contain the letter a.
+// b) Create the function that makes the test pass.
 
-// a) Create a test with an expect statement using the variable provided.
+// Create a method called aLetterFilter
+// Takes an array as a parameter
+const aLetterFilter = (array) => {
+// Create a new array to push the words into
+    let newArray = []
+// Iterate over the array using the .filter method to filter the correct words, and pass in a parameter of word
+    array.filter(word => {
+// Make a conditional, using the .includes method, check if the word includes a substring of 'a' or includes a substring of 'A', because the includes method is case-sensitive
+        if(word.includes('a') || word.includes('A')) {
```

Comment:
This works! but it's helpful to remember that .filter() has a built in conditional statement. On top of that it returns to us a new array. So filter is doing a lot of the work here for you that you are also doing. Conceptually on point though!